### DISCUSSION TITLE
Is that possible to test DI without running TestClient requests?

### DISCUSSION THREAD

**Discussion by AlexanderFarkas** [fastapi/fastapi#8804] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

class A:
    def __init__():
        print("Initialized")

my_dep = app.dependency(A)

# Expecting to see "Initialized" in console
```

### Description

Is that possible to test FastAPI DI without running TestClient? For example, test UseCase (clean architecture) without manually instantiating its dependencies.  

### Operating System

macOS

### FastAPI Version

0.68.0

### Python Version

3.9.5

**Comment by chbndrhnns** [comment, marked as answer]
In my understanding, the dependency resolution mechanism only takes place in the request/response cycle.

**Comment by adriangb** [comment]
It is not currently possible. This could be resolved by #3641
